src/mbgl/map/map.cpp

```diff
@@ -142,8 +142,14 @@ Map::Impl::Impl(Map& map_,
       mode(mode_),
       contextMode(contextMode_),
       pixelRatio(pixelRatio_),
-      asyncUpdate([this] { update(); }),
-      annotationManager(std::make_unique<AnnotationManager>(pixelRatio)) {
+      annotationManager(std::make_unique<AnnotationManager>(pixelRatio)),
+      asyncInvalidate([this] {
+          if (mode == MapMode::Continuous) {
+              backend.invalidate();
+          } else {
+              renderStill();
+          }
```

Comment:
Moved the invalidation call back to happen after the async. The change to master is now that before, we had a `render() => asyncUpdate() => update() => invalidate() => render() ` flow. It's now shortened to `render() => asyncInvalidate() => invalidate() => update+render()`